salt/modules/parted_partition.py

```diff
@@ -542,36 +545,19 @@ def mkpart(device, part_type, fs_type=None, start=None, end=None):
     return out
 
 
-def mkpartfs(device, part_type, fs_type, start, end):
+def mkpartfs(device, part_type, fs_type=None, start=None, end=None):
     """
-    Make a <part_type> partition with a new filesystem of <fs_type>, beginning
-    at <start> and ending at <end> (by default in megabytes).
-
-    <part_type> should be one of "primary", "logical", or "extended". <fs_type>
-    must be one of "ext2", "fat32", "fat16", "linux-swap" or "reiserfs" (if
-    libreiserfs is installed)
+    The mkpartfs actually is an alias to mkpart and is kept for compatibility.
+    To know the valid options and usage syntax read mkpart documentation.
 
-    CLI Example:
+    CLI Examples:
 
     .. code-block:: bash
 
-        salt '*' partition.mkpartfs /dev/sda logical ext2 440 670
+        salt '*' partition.mkpartfs /dev/sda primary fs_type=fat32 start=0 end=639
+        salt '*' partition.mkpartfs /dev/sda primary start=0 end=639
     """
-    _validate_device(device)
```

Comment:
thanks for updating this. looks like this is the difference between the two. Is there a reason we dont want to add this to `mkpart`?

Reply:
No reason at all. I just re-added the _validate_device and updated the tests accordingly.